app.py

```diff
@@ -13,29 +12,23 @@
 
 
 # Main function
-def nl_to_sql(prompt):
-    # Translate the natural language query to SQL using the OpenAI API
-    sql_query = utils.get_openai_response(prompt, "sql_agent")
-
-    # Process the SQL query
-    query_result = utils.process_query(sql_query)
+def nl_to_sql(prompt: str):
```

Comment:
change this method (or refactor to a _private method) such that it accepts `llm: langchain.llms.base.LLM`
And in the "production" code make sure to provide it with an OpenAI object, while in the unit testing code use https://github.com/hwchase17/langchain/blob/master/langchain/llms/fake.py